### PR TITLE
Disable employment and education delete buttons during API activity

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #1752

#### What's this PR do?
Disable employment and education delete buttons during API activity

### CODE REVIEW

**Comment by aliceriot** [comment]
When the deletion request is in-flight the 'Next' button at the bottom spins as well:

weird_spinner_thing

that was a difficult screenshot to grab! haha

**Comment by noisecapella** [comment]
Yeah, that happens with the pay buttons too. When you click one pay button they all have the spinner. I'm not sure what we can do here without increasing the complexity of the fetchStatus code. What do you think?

**Comment by aliceriot** [comment]
I think it could be fine for now, in the future each button could have a `UUID` which would be dispatched to the store on click, which would allow us to discern which button was clicked. Maybe we should file an issue and come back to that? That would be a bit of work to implement so I don't think it quite makes sense to try to solve that problem here.

File: static/js/components/EducationForm.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ class EducationForm extends ProfileFormFields {
   };
 
   deleteEducationEntry: Function = (): void => {
-    deleteEducationEntry.call(this);
+    return deleteEducationEntry.call(this);
```
Comment (aliceriot) [inline static/js/components/EducationForm.js]: is this because `deleteEducationEntry` returns a `Promise`?

Comment (aliceriot) [inline static/js/components/EducationForm.js]: (below for similar work history change too)

File: static/js/util/profile_history_edit.js [hunk 2]
```diff
@@ -79,5 +79,5 @@ export function deleteWorkHistoryEntry () {
   const { saveProfile, profile, ui } = this.props;
   let clone = _.cloneDeep(profile);
   clone['work_history'].splice(ui.deletionIndex, 1);
-  saveProfile(employmentValidation, clone, ui);
+  return saveProfile(employmentValidation, clone, ui);
```
Comment (aliceriot) [inline static/js/util/profile_history_edit.js]: ah ok I see, these functions were changed to return the `saveProfile` promise. looks good!

**Comment by noisecapella** [comment]
Sounds good, I created https://github.com/mitodl/micromasters/issues/2140

**Pull request merged**